### ISSUE TITLE
Implicit output port on p:if

### ISSUE THREAD

**Issue by xml-project** [xproc/3.0-specification#886]
Because I ran into this problem (and Erik found it): We should explicitly state, that the implicit output port of p:if has content-types='any' and is a sequence port.

**Comment by xml-project** [comment]
Consider the following pipeline fragment (delivered by @eriksiegel )
````
<p:for-each>
    <p:if test="p:document-property(., 'base-uri') eq 'demo.html'">
      <p:insert match="/html/body" position="last-child">
        <p:with-input port="insertion">
          <div>
            <p>This was inserted by our pipeline</p>
          </div>
        </p:with-input>
      </p:insert>
    </p:if>
</p:for-each>
````
The p:if does not specify a primary output port, so the specs says, one is implicitly constructed, but it does not say how. The implicitly constructed output port of a compound step is an anonymous port with `content-type` and `sequence` taken from the last step of the subpipeline. 
But this does not work here because it would result in `content-types='xml html'` (from p:insert). Every non-XML, non-HTML document processed by p:if (appearing on the implicit else) would raise an error.
Therefor: The implicit constructed primary output port must have `content-types='any'` and `sequence=true`. Right?
Then: What do we say, if an explicit primary output port is defined on p:if. Is it the authors duty to cover the if-case and the identity case with her definition? 
I think this is error prone. Shouldn't we say, that `content-types` and `sequence` from the explicitly declared output port are ignored?
What did I miss?

**Comment by eriksiegel** [comment]
I think you missed nothing. Probably constructed output ports must always have `sequence="true"` and `content-types="any"`

There's probably a catch somewhere but I don't see it (yet)

**Comment by xml-project** [comment]
Fixed with #888
